src/Microsoft.Azure.SignalR.Protocols/CheckWithAckMessage.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.SignalR.Protocol
+{
+    /// <summary>
+    /// Base class of check-with-ack messages between Azure SignalR Service and SDK.
+    /// </summary>
+    public abstract class CheckWithAckMessage : ExtensibleServiceMessage, IAckableMessage, IMessageWithTracingId
+    {
+        /// <summary>
+        /// Gets or sets the ack id.
+        /// </summary>
+        public int AckId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tracing Id
+        /// </summary>
+        public ulong? TracingId { get; set; }
+
+        protected CheckWithAckMessage(int ackId, ulong? tracingId)
+        {
+            AckId = ackId;
+            TracingId = tracingId;
+        }
+    }
+
+    /// <summary>
+    /// A waiting for ack check-user-in-group message.
+    /// </summary>
+    public class CheckUserInGroupWithAckMessage : CheckWithAckMessage
+    {
+        /// <summary>
+        /// Gets or sets the user Id.
+        /// </summary>
+        public string UserId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the group name.
+        /// </summary>
+        public string GroupName { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LeaveGroupWithAckMessage"/> class.
+        /// </summary>
+        /// <param name="userId">The connection Id.</param>
+        /// <param name="groupName">The group name, from which the connection will leave.</param>
```

Comment:
"from which the connection will leave" is a little confusing, could you provide more context?

Reply:
the grammer is right - connection will leave from the `group`

Reply:
I am not meaning the grammar, but the context of using the message. Is the message used only when a connection is going to leave a group? If so, I think pointing out the usage scenario in the comments would be better for code readers to understand.

Reply:
I am also wondering the usage scenario of `CheckWithAckMessage`s. Do I need to provide a user interface for SDK users to send this kind of messages?

Reply:
According to @vwxyzh , when to use `CheckWithAckMessage` depends on users, so this kind of messages is universal, and no need to add `from which the connection will leave.`

Reply:
I found the comments are simplyd copied by `LeaveGroupWithAckMessage` by previous author, need to update.